src/scanner.cc

```diff
@@ -905,12 +948,12 @@ class Scanner
                                                         '~', '$', '_', '^'};
     const std::unordered_map<char, TokenType> m_AttachedModifiers = {
         {'*', BOLD_OPEN},        {'/', ITALIC_OPEN},    {'-', STRIKETHROUGH_OPEN},
-        {'_', UNDERLINE_OPEN},   {'|', SPOILER_OPEN},   {'`', VERBATIM_OPEN},
+        {'_', UNDERLINE_OPEN},   {'!', SPOILER_OPEN},   {'`', VERBATIM_OPEN},
         {'^', SUPERSCRIPT_OPEN}, {',', SUBSCRIPT_OPEN}, {'+', INLINE_COMMENT_OPEN},
-        {'$', INLINE_MATH_OPEN}, {'=', VARIABLE_OPEN},
+        {'$', INLINE_MATH_OPEN}, {'=', VARIABLE_OPEN}, {'|', RANGED_MODIFIER_OPEN},
```

Comment:
We need to refactor the ranged modifiers slightly. Instead of having a single bitset we need one bitset per attached modifier. Otherwise we will not be able to nest multiple ranged modifiers into one another.